spe_report/summary.py

```diff
@@ -40,26 +43,45 @@ def email_report(self):
         useTestLastTakenTime: {self.used_query_date}
         updatedTestLastTakenTime: {self.next_stored_persisted_date}
         
+        
         course_id: {self.course_id}
         
-        user scores added: {len(self.sent_score_list)} failed scores addition: {len(self.not_sent_scores_list)}
          """
         return msg
 
     def email_msg(self):
-        msg: List[str] = [self.email_report()]
-        if self.sent_score_list:
-            msg.append("Success users List:")
-            for item in self.sent_score_list:
+        msg: List[str] = list()
+        msg.append(self.email_report())
+        msg.append(f"Placement user scores added: {len(self.sent_score_list[PLACEMENT])} failed: {len(self.not_sent_scores_list[PLACEMENT])}")
+        if self.sent_score_list[PLACEMENT]:
+            msg.append("Placement Success users List:")
+            for item in self.sent_score_list[PLACEMENT]:
+                msg.append(f"user: {item['user']}  score: {item['score']}  finished_at: {item['local_submitted_date']}")
+            msg.append("\n")
+        if self.not_sent_scores_list[PLACEMENT]:
+            msg.append("Failed sending Placement scores below, will try in the next run")
+            for item in self.not_sent_scores_list[PLACEMENT]:
                 msg.append(f"user: {item['user']}  score: {item['score']}  finished_at: {item['local_submitted_date']}")
             msg.append("\n")
-        if self.not_sent_scores_list:
-            msg.append("Failed to send users List below, will try in the next run")
-            for item in self.not_sent_scores_list:
+
+        msg.append(f"Validation user scores added: {len(self.sent_score_list[VALIDATION])} failed: {len(self.not_sent_scores_list[VALIDATION])}")
+        if self.sent_score_list[VALIDATION]:
+            msg.append("Validation Success users List:")
+            for item in self.sent_score_list[VALIDATION]:
                 msg.append(f"user: {item['user']}  score: {item['score']}  finished_at: {item['local_submitted_date']}")
+            msg.append("\n")
+        if self.not_sent_scores_list[VALIDATION]:
+            msg.append("Failed sending Validation scores below, will try in the next run")
+            for item in self.not_sent_scores_list[VALIDATION]:
+                msg.append(f"user: {item['user']}  score: {item['score']}  finished_at: {item['local_submitted_date']}")
+            msg.append("\n")
```

Comment:
The issue with fixing the repetition is in the email using a loop is, I am reporting all the placement related stuff first and then validation. Each exam type has 2 pieces of info to be said so repetition is not avoidable or I have to write some complicated logic. Here is the sample of the email report looks like 
````
'Placement user scores added: 6 failed: 2'
b'Placement Success users List:'
b'user: abc  score: 4935.3  finished_at: 2019-09-05 12:00:01-0400'
b'user: edf  score: 1525.0  finished_at: 2019-09-06 00:05:22-0400'
b'user: ert  score: 9.0  finished_at: 2019-09-06 12:30:36-0400'
b'user: fgh  score: 2423.2  finished_at: 2019-09-06 14:19:25-0400'
b'user: vvv  score: 208.0  finished_at: 2019-09-06 19:28:21-0400'
b'user: vde  score: 2325.2  finished_at: 2019-09-06 20:36:11-0400'
b'user: yyyy  score: 1826.0  finished_at: 2019-09-08 15:57:22-0400'

Placement Failure users List:
b'user: xswe  score: 3630.0  finished_at: 2019-09-08 22:38:46-0400'
b'user: rttggg  score: 1609.0  finished_at: 2019-09-12 10:04:42-0400'
b''
b''
b'Validation user scores added: 5 failed: 2'
b'Validation Success users List:'
b'user: wrfff  score: 3532.6  finished_at: 2019-10-16 17:54:57-0400'
b'user: ddddd  score: 2222.2  finished_at: 2019-10-16 17:56:32-0400'
b'user: ggggg  score: 4136.5  finished_at: 2019-10-16 17:57:17-0400'
b'user: ggggg  score: 4632.5  finished_at: 2019-10-16 17:57:57-0400'
b'user: bgbgb  score: 4334.3  finished_at: 2019-10-16 17:58:27-0400'
Validation Failure users List:'
b'user: hjmnh  score: 2830.0  finished_at: 2019-10-16 17:58:33-0400'
b'user: vfrtth  score: 4135.5  finished_at: 2019-10-16 18:05:50-0400'
```